src/Hack.sol

```diff
@@ -112,55 +112,23 @@ contract Vault {
 
 /* Exercise 5 */
 
-/// @dev One party chooses Head or Tail and sends 1 ETH.
-///      The next party sends 1 ETH and tries to guess what the first party chose.
-///      If they succeed, they get 2 ETH, else the first party gets 2 ETH.
-contract HeadTail {
-    address payable public partyA;
-    address payable public partyB;
-    bytes32 public commitmentA;
-    bool public chooseHeadB;
-    uint256 public timeB;
-
-    /* CONSTRUCTOR */
-
-    /// @param _commitmentA is the result of the following command: keccak256(abi.encode(chooseHead,randomNumber)).
-    constructor(bytes32 _commitmentA) payable {
-        require(msg.value == 1 ether);
-
-        commitmentA = _commitmentA;
-        partyA = payable(msg.sender);
-    }
-
-    /// @dev Guesses the choice of party A.
-    /// @param _chooseHead True if the guess is Head, false otherwise.
-    function guess(bool _chooseHead) public payable {
-        require(msg.value == 1 ether);
-        require(partyB == address(0));
-
-        chooseHeadB = _chooseHead;
-        timeB = block.timestamp;
-        partyB = payable(msg.sender);
-    }
-
-    /// @dev Reveals the committed value and sends ETH to the winner.
-    /// @param _chooseHead True if Head was chosen, false otherwise.
-    /// @param _randomNumber The random number chosen to obfuscate the commitment.
-    function resolve(bool _chooseHead, uint256 _randomNumber) public {
-        require(msg.sender == partyA);
-        require(keccak256(abi.encode(_chooseHead, _randomNumber)) == commitmentA);
-        require(address(this).balance >= 2 ether);
-
-        if (_chooseHead == chooseHeadB) partyB.transfer(2 ether);
-        else partyA.transfer(2 ether);
-    }
-
-    /// @dev Time out party A if it takes more than 1 day to reveal.
-    ///      Sends ETH to party B.
-    function timeOut() public {
-        require(block.timestamp > timeB + 1 days);
-        require(address(this).balance >= 2 ether);
-        partyB.transfer(2 ether);
+/// @dev Contract for locking and unlocking funds using a commitment and password.
+contract Locker {
+    bytes32 commitment;
+
+    /// @dev Locks the funds by setting the commitment.
+    /// @param _commitment The commitment to lock the funds.
+    function lock(bytes32 _commitment) external payable {
```

Comment:
```suggestion
    function lock(bytes32 commitment) external payable {
```

we could move to this pattern tbh (to do in another PR)

Reply:
the problem is that the storage variable is also named `commitment`

Reply:
Oh yes ok I understand